packages/jobs/lib/execution/action.ts

```diff
@@ -0,0 +1,221 @@
+
+        providerConfig = await configService.getProviderConfig(task.connection.provider_config_key, task.connection.environment_id);
+        if (providerConfig === null) {
+            throw new Error(`Provider config not found for connection: ${task.connection.connection_id}`);
+        }
+
+        const syncConfig = await getSyncConfigRaw({
+            environmentId: providerConfig.environment_id,
+            config_id: providerConfig.id!,
+            name: task.actionName,
+            isAction: true
+        });
+        if (!syncConfig) {
+            throw new Error(`Action config not found: ${task.id}`);
+        }
+
+        const logCtx = await logContextGetter.get({ id: String(task.activityLogId) });
+
+        const nangoProps: NangoProps = {
+            scriptType: 'action',
+            host: getApiUrl(),
+            teamId: account.id,
+            teamName: account.name,
+            connectionId: task.connection.connection_id,
+            environmentId: task.connection.environment_id,
+            environmentName: environment.name,
+            providerConfigKey: task.connection.provider_config_key,
+            provider: providerConfig.provider,
+            activityLogId: logCtx.id,
+            secretKey: environment.secret_key,
+            nangoConnectionId: task.connection.id,
+            attributes: syncConfig.attributes,
+            syncConfig: syncConfig,
+            debug: false,
+            runnerFlags: await getRunnerFlags(),
+            startedAt: new Date()
+        };
+
+        metrics.increment(metrics.Types.ACTION_EXECUTION, 1, { accountId: account.id });
+
+        const res = await startScript({
+            taskId: task.id,
+            nangoProps,
+            logCtx: logCtx,
+            input: task.input
+        });
+
+        if (res.isErr()) {
+            throw res.error;
+        }
+
+        return Ok(undefined);
+    } catch (err) {
+        const error = new NangoError('action_script_failure', { error: err instanceof Error ? err.message : err });
+        await onFailure({
+            connection: {
+                id: task.connection.id,
+                connection_id: task.connection.connection_id,
+                environment_id: task.connection.environment_id,
+                provider_config_key: task.connection.provider_config_key
+            },
+            syncName: task.actionName,
+            provider: providerConfig?.provider || 'unknown',
+            providerConfigKey: task.connection.provider_config_key,
+            activityLogId: task.activityLogId,
+            runTime: 0,
+            error,
+            environment: { id: task.connection.environment_id, name: environment?.name || 'unknown' },
+            ...(account?.id && account?.name ? { team: { id: account.id, name: account.name } } : {})
+        });
+        return Err(error);
+    }
+}
+export async function handleActionOutput({ nangoProps }: { nangoProps: NangoProps }): Promise<void> {
```

Comment:
I would have named handle*Success to be consistent with handle*Error

Reply:
good idea